decorator.rb

```diff
@@ -0,0 +1,37 @@
+require './nameable'
+require './person'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name.upcase
```

Comment:
- [ ] It would be better if we change the `upcase` to `capitalize`, capitalize takes the first letter and capitalize it and leave the remaining letters the way it is whilst the upcase capitalize all the letters